### PR TITLE
-chests-all-monsters,  -loot,  and -no-random-encounters

### PR DESCRIPTION
**Pull request by HansGR**
Three new flags:
--chests-all-monsters,  usage -cam <boss%>:  replaces all chests with Monsters-in-a-box, with <boss%> = [0--100] percentage of being a boss.  Non-bosses include normal MiaBs and randomly selected formations.
--loot,  usage -loot:  randomizes drops and steals for all enemies.  No control flags currently, 100% random from all items.
--no-random-encounters,  usage -nre:  sets encounter rate to 0 in all maps.  Changes Moogle Charm to Moogle Curse: when equipped, allows encounters on world map only at a low rate.

### CODE REVIEW

File: data/data.py [hunk 1]
```diff
@@ -38,8 +38,11 @@ def __init__(self, rom, args):
         self.maps = maps.Maps(rom, args, self.items)
         self.maps.mod(self.characters)
 
-        self.enemies = enemies.Enemies(rom, args)
+        self.enemies = enemies.Enemies(rom, args, self.items)
         self.enemies.mod(self.maps)
+        #for p in self.enemies.packs.packs:
+        #    forms = [(f, self.enemies.formations.get_name(f)) for f in p.formations]
+        #    print(p.id, ': ', forms)
```
Comment (asilverthorn) [inline data/data.py]: minor nit: please remove commented out code

File: data/items.py [hunk 2]
```diff
@@ -187,6 +188,20 @@ def moogle_starting_equipment(self):
...
+        moogle_charm = self.items[name_id["Moogle Charm"]]
+        moogle_charm.name = "Moogle Curse"
+
+        descr_offset = self.rom.get_bytes(0x2d7aa0 + moogle_charm.id * 2, 2)
+        offset = descr_offset[1] * 0x100 + descr_offset[0]
+
+        # new_descr = "No random enemy encounters<end>"
+        new_descr = "Draw monsters on world map<end>"
+        name_bytes = text.get_bytes(new_descr, text.TEXT2)
+        self.rom.set_bytes(0x2d6400 + offset, name_bytes)
```
Comment (asilverthorn) [inline data/items.py]: Please structure the setting of an item description like this (make a desc_data array in items.py and set the description in item.py):

https://github.com/ff6wc/WorldsCollide/pull/28/files#diff-5991f3562b7b58a2c5931c6a2897859699dea4634f1e33cd7511655c86c1165d

File: event_battle_groups.txt [hunk 3]
```diff
@@ -0,0 +1,83 @@
+Formations to avoid:
```
Comment (asilverthorn) [inline event_battle_groups.txt]: is this file needed? Looks like it's the same data as enemy_battle_groups.py

Comment (HansGR) [reply event_battle_groups.txt]: This file is not needed. I don't know how to remove it from the branch though. Do I just delete the file and then push a new commit?

Comment (HansGR) [reply event_battle_groups.txt]: OK, I removed it.

**Comment by HansGR** [comment]
All comments addressed.  -DT

**Comment by HansGR** [comment]
I changed -loot to -ssd <random%> (--shuffle-steals-drops).  This makes it more useful in general and follows behavior of shuffle chests random%.  

I also fixed the tracking menu entry for -cam, which was broken.

File: args/steal.py [hunk 4]
```diff
@@ -20,19 +27,27 @@ def flags(args):
...
         steal_chances = "Higher"
     if args.steal_chances_always:
         steal_chances = "Always"
 
-    return [
-        ("Chances", steal_chances),
-    ]
+    result.append(("Chances", steal_chances))
+
+    result.append(("Shuffle steals & drops", args.shuffle_steals_drops))
```
Comment (asilverthorn) [inline args/steal.py:46]: too long:

image

File: args/steal.py [hunk 5]
```diff
@@ -20,19 +27,27 @@ def flags(args):
...
+    if args.shuffle_steals_drops:
+        result.append(("Random Percent:", f"{args.shuffle_steals_drops_random_percent}%"))
```
Comment (asilverthorn) [inline args/steal.py:48]: nit: please remove the ":" to match others

**Comment by asilverthorn** [comment]
Partially (all but no-random-encounters) merged with 1.3.0 https://github.com/ff6wc/WorldsCollide/pull/64
